suite-native/module-send/src/hooks/useCustomFee.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useCallback, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { isRejected } from '@reduxjs/toolkit';
+import { G } from '@mobily/ts-belt';
+
+import { AccountsRootState, selectAccountNetworkType } from '@suite-common/wallet-core';
+import {
+    AccountKey,
+    isFinalPrecomposedTransaction,
+    TokenAddress,
+} from '@suite-common/wallet-types';
+import { useTranslate } from '@suite-native/intl';
+import { useDebounce } from '@trezor/react-utils';
+import { BigNumber } from '@trezor/utils';
+import { useFormContext } from '@suite-native/forms';
+
+import { SendFeesFormValues } from '../sendFeesFormSchema';
+import {
+    selectCustomFeeLevel,
+    NativeSendRootState,
+    selectFeeLevelTransactionBytes,
+} from '../sendFormSlice';
+import { calculateCustomFeeLevelThunk } from '../sendFormThunks';
+
+type UseCustomFeeProps = {
+    accountKey: AccountKey;
+    tokenContract?: TokenAddress;
+};
+
+export const useCustomFee = ({ accountKey, tokenContract }: UseCustomFeeProps) => {
+    const dispatch = useDispatch();
+    const debounce = useDebounce();
+    const { translate } = useTranslate();
+
+    const [isErrorBoxVisible, setIsErrorBoxVisible] = useState(false);
+    const [isFeeLoading, setIsFeeLoading] = useState(false);
+
+    const customFeeLevel = useSelector(selectCustomFeeLevel);
+    const networkType = useSelector((state: AccountsRootState) =>
+        selectAccountNetworkType(state, accountKey),
+    );
+
+    const {
+        formState: { errors },
+        setError,
+        trigger,
+        getValues,
+        watch,
+    } = useFormContext<SendFeesFormValues>();
+
+    const feePerUnitFieldName = 'customFeePerUnit';
+    const feeLimitFieldName = 'customFeeLimit';
+
+    const watchedFeePerUnit = watch(feePerUnitFieldName, '0');
+    const watchedFeeLimit = watch(feeLimitFieldName, '1') as string;
+
+    const normalLevelTransactionBytes = useSelector((state: NativeSendRootState) =>
+        selectFeeLevelTransactionBytes(state, 'normal'),
+    );
+
+    const handleValuesChange = useCallback(async () => {
+        const { customFeePerUnit, customFeeLimit } = getValues();
+
+        trigger();
```

Comment:
what does trigger do? shoudn't we trigger submit after following condition?

Reply:
The `trigger` function validates the form fields. If an error is found, it is displayed to the user. Ideally, we would not have to call `trigger` empirically but rather use the `mode: 'onChange'` validation mode of the react-hook-form, so the library itself handles the validation. But we want to validate on `debounce()` so it is not validated while the user is still typing. For this reason, the trigger is called in the debounce onChange function.